File: packages/navigation/src/MarsRover/Navigation/LandRover.php
<?php

namespace MarsRover\Navigation;

use MarsRover\Geolocation\{
    Coordinates,
    Location,
    Orientation
};

class LandRover
{
    private $location;

    public function __construct($x, $y, $orientation)
    {
        $this->location = new Location(
            new Coordinates($x, $y),
            new Orientation($orientation)
        );
    }

    public function getLocation(): Location
    {
        return $this->location;
    }
}

File: kotlin/packages/navigation/src/main/kotlin/marsrover/navigation/LandRover.kt
package marsrover.navigation

import marsrover.eventsourcing.Event
import marsrover.eventsourcing.EventStore
import marsrover.geolocation.Coordinates
import marsrover.geolocation.Location
import marsrover.geolocation.Orientation
import java.time.Clock
import java.time.Instant

data class LandRover(private val x: Int, private val y: Int, private val orientation: Orientation) {

    val location = Location(Coordinates(x, y), orientation)
}

class LandRoverHandler(private val clock: Clock, private val eventStore: EventStore) {

    fun handle(landRover: LandRover) {
        val (coordinates, orientation) = landRover.location
        val roverLanded = RoverLanded(coordinates.x, coordinates.y, orientation, receivedAt = clock.instant())

        eventStore.log(roverLanded)
    }
}

data class RoverLanded(
    val x: Int,
    val y: Int,
    val orientation: Orientation,
    override val receivedAt: Instant
) : Event(receivedAt)
